proposals/0340-validate-chained-block-id.md

```diff
@@ -0,0 +1,67 @@
+---
+simd: '0340'
+title: Validate chained block id
+authors:
+  - Charles Li
+category: Standard
+type: Core
+status: Review
+created: 2025-08-20
+feature: TBD
+---
+
+## Summary
+
+Verify chained merkle roots across slot boundaries.
+
+## Motivation
+
+Currently it is not required to verify that a FEC set merkle root chains
+correctly across slot boundaries. Consensus can converge on a block even
+if the first FEC set's chained merkle root is invalid ie. does not chain
+off the parent block's last FEC set merkle root (the block id).
+
+This is a problem because chained merkle roots should validate an entire
+ancestry, so that you have a canonical linear chain (all the way back to
+the snapshot slot - see also SIMD-0333 proposal for including a block_id
+in the snapshot manifest). Otherwise, you don't know if your parent slot
+based on `slot - parent_off` is in fact your actual parent block because
+slot numbers do not key blocks uniquely when there is equivocation.
+
+## New Terminology
+
+N/A
+
+## Detailed Design
+
+Verify the chained merkle root of the first FEC set in a block correctly
+matches the merkle root of the last FEC set of the parent slot (based on
+the slot and parent_off field in every shred in the FEC set).
+
+If they do not match, mark the child block (ie. block with the first FEC
```

Comment:
to be extra clear I would say "do not attempt to replay a child block off a parent block unless BOTH the parent slot # and parent block id match"

Then in the section below we can say that for TowerBFT we mark it as dead (poh ticks already does this) so that repair can be notified. In Alpenglow we don't need any special handling, certificate conditions will notify repair.